src/utils.ts

```diff
@@ -61,3 +61,51 @@ export const svgToUrl = (icon: string) => {
   const base64SVG = window.btoa(icon);
   return `data:image/svg+xml;base64,${base64SVG}`;
 };
+
+
+
+export function useStorage<T>(
+  key: string,
+  initialValue: T,
+  persistent: boolean = true,
+): {
+  value: T;
+  setValue: (params: any) => void;
+  resetValue: () => void;
+} {
+  const storage = localStorage;
+
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    const item = storage.getItem(key);
+    return item ? JSON.parse(item) : initialValue;
+  });
+
+  useEffect(() => {
+    storage.setItem(key, JSON.stringify(storedValue));
+  }, [key, storedValue, storage]);
+
+  useEffect(() => {
+    const handleBeforeUnload = () => {
+      if (!persistent) {
+        setStoredValue(initialValue);
+      }
+    };
+    window.addEventListener('beforeunload', handleBeforeUnload);
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, [initialValue, persistent]);
```

Comment:
Gal geriau daryti on mount (paliekant dependency array tuscia)? Kadangi dabar kiekviena kart renderinant komponenta kur sitas hookas naudojamas, tai galimai is naujo settints listeneri. Objektas tarkim gali nesikeisti, bet objekto reference pasikeistir ir to pasekoje bus trgerintas sitas useEffect

Reply:
O gal isvis be event listenerio galima issprest issaugojima.

Reply:
Nu jo galiu palikti tuščia, bet tuščias dependency array nekels warning'ų libe?

Čia realiai , kad refreshinus puslapį, kartais reikia, kad neišsisaugotų filtrai (Jaroslavas sakė, nori kad nesaugotų). O be listenerio kazi ar imanoma nuvalyti filtrus? Galvojau kita storage naudoti, kaip session bet irgi refreshinus nenusivalo .